tests/010-test-pool-minimum-data-size.sh

```diff
@@ -0,0 +1,61 @@
+source $SRCDIR/libtest.sh
+
+# Test that the miminimum pool size is correctly 
+# allocated if the volume group created wouldn't 
+# be big enough.
+test_pool_minimum_data_size() {
+
+  local devs=$TEST_DEVS
+  local test_status
+  local testname=`basename "$0"`
+  local vg_name
+  local vg_name="dss-test-foo"
+  local size=0
+  local size_type="B"
+  local test_status
+
+ # Error out if any pre-existing volume group vg named dss-test-foo
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      echo "ERROR: $testname: Volume group $vg_name already exists."
+      return 1
+    fi
+  done
+  
+  test_status=1
+ 
+  # get total size of all block devices, since
+  # the dss partitions the PV's using 100% of the size
+  # of each device  
+  for d in $devs; do
+    size=$(( vg_size + `blockdev --getsize64 $d` ));
+  done
+
+  # Create config file
+  clean_config_files
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+MIN_DATA_SIZE="${size}B"
+EOF
+
+  # Run docker-storage-setup
+  $DSSBIN >> $LOGS 2>&1
+  
+  # Binary should fail, since there shouldn't be enough space for 
+  # the total size of all block devices to be
+  # used because dss sets aside ~0.1% for docker metadata.
+  # Failing to run binary should imply that d-s-s accounted 
+  # for user-specified MIN_DATA_SIZE.
+  # TODO: check for error message to get specific error?
+  if [ "$?" -eq 0 ]; then
+    cleanup $vg_name "$devs"
+    return 1
+  fi
+
+  test_status=0
```

Comment:
dss can fail for so many reasons and writing test cases for failure is hard. How about if we capture output and grep of message which says that minimum free space is not there. That way there is higher  certainty  that dss failed because minimum free space was not there and not for some other reason.